exporter/datadogexporter/internal/logs/translator_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+func TestTransform(t *testing.T) {
+	traceID := [16]byte{0x08, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x0, 0x0, 0x0, 0x0, 0x0a}
+	var spanID [8]byte
+	copy(spanID[:], traceID[8:])
+	ddTr := traceIDToUint64(traceID)
+	ddSp := spanIDToUint64(spanID)
+
+	type args struct {
+		lr  plog.LogRecord
+		res pcommon.Resource
+	}
+	tests := []struct {
+		name string
+		args args
+		want datadogV2.HTTPLogItem
+	}{
+		{
+			name: "log_with_attribute",
+			args: args{
+				lr: func() plog.LogRecord {
+					l := plog.NewLogRecord()
+					l.Attributes().InsertString("app", "test")
+					l.SetSeverityNumber(5)
+					return l
+				}(),
+				res: pcommon.NewResource(),
+			},
+			want: datadogV2.HTTPLogItem{
+				Message: *datadog.PtrString(""),
+				AdditionalProperties: map[string]string{
+					"app":              "test",
+					"status":           "debug",
+					otelSeverityNumber: "5",
+				},
+			},
+		},
```

Comment:
I think we could benefit from some more test cases narrowly targeting the logic for severity number/text since it's a bit tricky. For example:

- Severity number should take priority over severity text
- Severity text should be used for status if there is no severity number
- Severity text and severity number should be preserved regardless of which one is used for status

Having small test cases that specifically target these behaviors will help document them and make sure they aren't broken by accident in the future.

Reply:
Per my other comment, I'm no longer sure Severity number should take priority over severity text. But the point about adding the test cases remains.

Reply:
this has been fixed